nemo_curator/utils/script_utils.py

```diff
@@ -13,335 +13,1276 @@
+    ):
+        self.parser.add_argument(
+            "--output-dir",
+            type=str,
+            required=required,
+            help=help,
+        )
+
+    def add_arg_output_file_type(
+        self,
+        choices=None,
+        help="File type the dataset will be written to. Supported file formats "
+        "include 'jsonl' (default), 'pickle', or 'parquet'.",
+    ):
+        self.parser.add_argument(
+            "--output-file-type",
+            type=str,
+            default="jsonl",
+            choices=choices,
+            help=help,
+        )
+
+    def add_arg_output_train_file(self, help: str, default: str = None):
+        self.parser.add_argument(
+            "--output-train-file",
+            type=str,
+            default=default,
+            help=help,
+        )
+
+    def add_arg_protocol(self):
+        self.parser.add_argument(
+            "--protocol",
+            type=str,
+            default="ucx",
+            help="Protcol to use for dask cluster. "
+            "Note: This only applies to the localCUDACluster. If providing an user created "
+            "cluster refer to "
+            "https://docs.rapids.ai/api/dask-cuda/stable/api.html#cmdoption-dask-cuda-protocol",  # noqa: E501
+        )
+
+    def add_arg_rmm_pool_size(self):
+        self.parser.add_argument(
+            "--rmm-pool-size",
+            type=str,
+            default="14GB",
+            help="Initial pool size to use for the RMM Pool Memory allocator. "
+            "Note: This only applies to the localCUDACluster. If providing an user created "
+            "cluster refer to "
+            "https://docs.rapids.ai/api/dask-cuda/stable/api.html#cmdoption-dask-cuda-rmm-pool-size",  # noqa: E501
+        )
+
+    def add_arg_scheduler_address(self):
+        self.parser.add_argument(
+            "--scheduler-address",
+            type=str,
+            default=None,
+            help="Address to the scheduler of a created dask cluster. If not provided"
+            "a single node LocalCUDACluster will be started.",
+        )
+
+    def add_arg_scheduler_file(self):
+        self.parser.add_argument(
+            "--scheduler-file",
+            type=str,
+            default=None,
+            help="Path to the scheduler file of a created dask cluster. If not provided"
+            " a single node LocalCUDACluster will be started.",
+        )
+
+    def add_arg_seed(
+        self,
+        default=42,
+        help: str = "If specified, the random seed used for shuffling.",
+    ):
+        self.parser.add_argument(
+            "--seed",
+            type=int,
+            default=default,
+            help=help,
+        )
+
+    def add_arg_set_torch_to_use_rmm(self):
+        self.parser.add_argument("--set-torch-to-use-rmm", action="store_true")
+
+    def add_arg_shuffle(self, help: str):
+        ArgumentHelper.__attach_bool_arg(
```

Comment:
I believe this function call is missing an argparse object. Though, fixing it might be unnecessary with my other suggestions.

Reply:
Done! Thank you!